Align semantics and maction with MathML Core

eerii opened the pull request:
#### d133d2f62153c9d5d574827111eb0fbe53b2f36e
<pre>
Align semantics and maction with MathML Core
<a href="https://bugs.webkit.org/show_bug.cgi?id=262696">https://bugs.webkit.org/show_bug.cgi?id=262696</a>

Reviewed by Frédéric Wang.

The MathML Core spec says that semantics
(<a href="https://w3c.github.io/mathml-core/#semantics-and-presentation)">https://w3c.github.io/mathml-core/#semantics-and-presentation)</a> and
maction (<a href="https://w3c.github.io/mathml-core/#dfn-maction)">https://w3c.github.io/mathml-core/#dfn-maction)</a> have the same
layout algorithm as the mrow element. Additionally, add some rules to
the UA stylesheet to properly render these elements
(<a href="https://w3c.github.io/mathml-core/#user-agent-stylesheet).">https://w3c.github.io/mathml-core/#user-agent-stylesheet).</a>

Imports the /mathml/relations/html5-tree/maction-click.html test from
the WPT upstream repository.

Upstream commit: web-platform-tests/wpt@34e2cc1

Tests: imported/w3c/web-platform-tests/mathml/relations/html5-tree/maction-click-ref.html
       imported/w3c/web-platform-tests/mathml/relations/html5-tree/maction-click.html
       mathml/non-core/maction-dynamic.html
       mathml/non-core/maction-toggle-href.html
       mathml/non-core/maction-toggle.html
       mathml/non-core/maction.html
       mathml/non-core/semantics-2.html
       mathml/non-core/semantics-3.html
       mathml/non-core/semantics-4.html
       mathml/non-core/semantics.html
* LayoutTests/accessibility/math-foreign-content.html:
* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/html5-tree/maction-click-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/html5-tree/maction-click-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/html5-tree/maction-click.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/html5-tree/w3c-import.log:
* LayoutTests/mathml/non-core/maction-dynamic-expected.html: Renamed from LayoutTests/mathml/presentation/maction-dynamic-expected.html.
* LayoutTests/mathml/non-core/maction-dynamic.html: Renamed from LayoutTests/mathml/presentation/maction-dynamic.html.
* LayoutTests/mathml/non-core/maction-expected.html: Renamed from LayoutTests/mathml/presentation/maction-expected.html.
* LayoutTests/mathml/non-core/maction-toggle-expected.html: Renamed from LayoutTests/mathml/presentation/maction-toggle-expected.html.
* LayoutTests/mathml/non-core/maction-toggle-href-expected.html: Renamed from LayoutTests/mathml/presentation/maction-toggle-href-expected.html.
* LayoutTests/mathml/non-core/maction-toggle-href.html: Renamed from LayoutTests/mathml/presentation/maction-toggle-href.html.
* LayoutTests/mathml/non-core/maction-toggle.html: Renamed from LayoutTests/mathml/presentation/maction-toggle.html.
* LayoutTests/mathml/non-core/maction.html: Renamed from LayoutTests/mathml/presentation/maction.html.
* LayoutTests/mathml/non-core/semantics-2-expected.html: Renamed from LayoutTests/mathml/presentation/semantics-2-expected.html.
* LayoutTests/mathml/non-core/semantics-2.html: Renamed from LayoutTests/mathml/presentation/semantics-2.html.
* LayoutTests/mathml/non-core/semantics-3-expected.html: Renamed from LayoutTests/mathml/presentation/semantics-3-expected.html.
* LayoutTests/mathml/non-core/semantics-3.html: Renamed from LayoutTests/mathml/presentation/semantics-3.html.
* LayoutTests/mathml/non-core/semantics-4-expected.html: Renamed from LayoutTests/mathml/presentation/semantics-4-expected.html.
* LayoutTests/mathml/non-core/semantics-4.html: Renamed from LayoutTests/mathml/presentation/semantics-4.html.
* LayoutTests/mathml/non-core/semantics-expected.html: Renamed from LayoutTests/mathml/presentation/semantics-expected.html.
* LayoutTests/mathml/non-core/semantics.html: Renamed from LayoutTests/mathml/presentation/semantics.html.
* LayoutTests/platform/glib/accessibility/math-foreign-content-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/html/syntax/parsing/html-integration-point-expected.txt: Added.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/presentation-markup/mrow/legacy-mrow-like-elements-001-expected.txt: Added.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/relations/css-styling/default-properties-on-semantics-and-maction-expected.txt: Added.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/relations/css-styling/ignored-properties-001-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/relations/css-styling/writing-mode/writing-mode-002-expected.txt: Added.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/relations/html5-tree/dynamic-childlist-002-expected.txt:
* LayoutTests/platform/glib/mathml/presentation/foreign-element-in-annotation-xml-expected.txt: Added.
* LayoutTests/platform/mac/accessibility/math-foreign-content-expected.txt:
* LayoutTests/svg/outermost-svg-root.html:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/css/mathmlCoreExtras.css: Added.
(semantics &gt; :not(:first-child)):
(maction &gt; :not(:first-child)):
* Source/WebCore/mathml/MathMLAnnotationElement.cpp:
(WebCore::MathMLAnnotationElement::createElementRenderer):
(WebCore::MathMLAnnotationElement::childShouldCreateRenderer const):
* Source/WebCore/mathml/MathMLSelectElement.cpp:
(WebCore::MathMLSelectElement::childShouldCreateRenderer const):
(WebCore::MathMLSelectElement::updateSelectedChild):
(WebCore::MathMLSelectElement::defaultEventHandler):
(WebCore::MathMLSelectElement::willRespondToMouseClickEventsWithEditability const):
* Source/WebCore/style/InspectorCSSOMWrappers.cpp:
(WebCore::Style::InspectorCSSOMWrappers::collectDocumentWrappers):
* Source/WebCore/style/UserAgentStyle.cpp:
(WebCore::Style::UserAgentStyle::ensureDefaultStyleSheetsForElement):
* Source/WebCore/style/UserAgentStyle.h:

Canonical link: <a href="https://commits.webkit.org/300875@main">https://commits.webkit.org/300875@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d3c6c5b191fdf5a19e1827e03ab38cd1f1b7801e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/124178 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/43871 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/34588 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131005 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/76269 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/126055 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/44612 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/52471 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/94459 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/62663 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127132 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/35551 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111076 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75050 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/34495 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/29238 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/74491 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105292 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/29459 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/133684 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51101 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/38947 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/102933 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/51487 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/107294 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/102739 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26133 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48095 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/26344 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/47992 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/50962 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/56737 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/50409 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/53761 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52085 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

fred-wang commented on Source/WebCore/css/mathml.css:
```diff
@@ -94,40 +94,30 @@ math[display="inline" i] {
-
-mfrac > * {
-    math-style: compact;
+/* <mrow>-like elements */
```
Can we please handle the CSS changes unrelated to semantics/maction in a separate bug/patch?

Also, I'm a bit concern that the CSS changes for semantics/maction are not gated by the MathMLCoreEnabled flag, so this is effectively changing existing behavior.

In Gecko it was possible to use some media-query for that purpose ( see https://phabricator.services.mozilla.com/D49285#change-3XoUbGliyRXY ) but I'm not sure WebKit provides something like that currently (I guess we can introduce one though). I also don't see an easy way to load the a CSS UA sheet conditionally.

eerii replied:
I opened #51489 with the other CSS changes.

That is a good point, thank you for bringing it up. I asked about this and a way to to do it is to create a different stylesheet (i.e. `mathmlCore.css`) and include it conditionally. I don't think media-queries like Gecko are supported (if they are that would be great).

eerii replied:
Reviewing the point about gating the CSS changes, the failures in the tests for iOS/Mac are actually because of that

fred-wang commented on Source/WebCore/mathml/MathMLSelectElement.cpp:
```diff
@@ -217,7 +220,7 @@ void MathMLSelectElement::updateSelectedChild()
 void MathMLSelectElement::defaultEventHandler(Event& event)
 {
-    if (isAnyClick(event)) {
+    if (!document().settings().coreMathMLEnabled() && isAnyClick(event)) {
```
I wonder if we have WPT tests for this case (verifying that maction with toggle does not react to click events).

eerii replied:
I was searching and I don't think we have. I can make one to go along with this patch

fred-wang replied:
Yes please. I think that would be nice to have.

eerii replied:
Done, I'll upload it in the next commit with the changes to the stylesheet.

fred-wang commented on LayoutTests/imported/w3c/web-platform-tests/mathml/relations/html5-tree/maction-element-click-expected.png:
WPT reftests don't need PNG expectations: https://web-platform-tests.org/writing-tests/reftests.html

eerii replied:
Sorry about that

fred-wang commented on LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/relations/html5-tree/maction-element-click-expected.txt:
WPT reftests shouldn't have the WebKit's render tree reference.

fred-wang commented on Source/WebCore/CMakeLists.txt:
```diff
@@ -2084,6 +2084,7 @@ set(WebCore_USER_AGENT_STYLE_SHEETS
     ${WEBCORE_DIR}/css/html.css
     ${WEBCORE_DIR}/css/htmlSwitchControl.css
     ${WEBCORE_DIR}/css/mathml.css
+    ${WEBCORE_DIR}/css/mathmlCore.css
```
nit: I wonder if it should be called `mathmlCoreExtras.css`, since mathml.css actually has most of the CSS rules from MathML Core.

fred-wang commented on LayoutTests/imported/w3c/web-platform-tests/mathml/relations/html5-tree/maction-element-click-expected.html:
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
```
I don't remember what are the rules for -expected.html and -ref.html these days in WebKit. But please make sure we only export the -ref.html one when this is upstreamed to WPT.

fred-wang commented on LayoutTests/imported/w3c/web-platform-tests/mathml/relations/html5-tree/maction-element-click.html:
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE html>
```
I see we force MathML to be enabled in some cases, so probably we can do that here too: https://searchfox.org/wubkat/search?q=CoreMathMLEnabled%3Dtrue&path=LayoutTests%2Fimported%2Fw3c%2Fweb-platform-tests%2Fmathml%2F&case=true&regexp=false

eerii replied:
I created https://github.com/web-platform-tests/wpt/pull/55173 since the exporter doesn't clean up the comments. Once it is synced I'll update the expected test here to use `CoreMathMLEnabled`.

fred-wang commented on LayoutTests/platform/ios/TestExpectations:
```diff
@@ -7524,6 +7524,7 @@ imported/w3c/web-platform-tests/mathml/presentation-markup/mrow/inferred-mrow-st
 imported/w3c/web-platform-tests/mathml/presentation-markup/operators/embellished-op-4-1.html [ ImageOnlyFailure ]
 imported/w3c/web-platform-tests/mathml/presentation-markup/operators/embellished-op-4-2.html [ ImageOnlyFailure ]
 imported/w3c/web-platform-tests/mathml/presentation-markup/operators/embellished-op-4-3.html [ ImageOnlyFailure ]
+imported/w3c/web-platform-tests/mathml/relations/html5-tree/maction-element-click.html [ ImageOnlyFailure ]
```
That won't be necessary if you force MathMLCore to be enabled (also, probably other non-Apple platforms with MathMLCore disabled by default would fail otherwise).

fred-wang commented on LayoutTests/platform/mac/TestExpectations:
```diff
@@ -1643,6 +1643,7 @@ media/media-source/media-source-unnecessary-seek-seeked.html [ Pass Failure ]
 # imported/w3c/web-platform-tests/mathml/presentation-markup/operators/mo-stretch-properties-dynamic-001.html is flaky
 webkit.org/b/228094 imported/w3c/web-platform-tests/mathml/presentation-markup/operators/mo-stretch-properties-dynamic-001.html [ Pass Failure ]
 
+imported/w3c/web-platform-tests/mathml/relations/html5-tree/maction-element-click.html [ ImageOnlyFailure ]
```
Same here.

fred-wang commented on LayoutTests/svg/outermost-svg-root.html:
```diff
@@ -1,4 +1,4 @@
-<!DOCTYPE html>
+<!DOCTYPE html><!-- webkit-test-runner [ CoreMathMLEnabled=false ] -->
```
I don't understand why this is necessary? `annotation-xml` is the first child, so it should be rendered already? 

Skimming over https://bugs.webkit.org/show_bug.cgi?id=223539 it does not seem the semantics element is important at all. So maybe just `<math><mtext><svg></svg></mtext></math>` would work here.

(incidentally, `let svg = document.getElementById(id).firstElementChild.cloneNode(true);` below should probably renamed `clonedChild` or something, since it's a math tag in that case)

eerii replied:
I changed it to `mtext` and renamed the variable, it seems to work now.

fred-wang commented on LayoutTests/accessibility/math-foreign-content.html:
```diff
@@ -1,4 +1,4 @@
-<!DOCTYPE HTML>
+<!DOCTYPE HTML><!-- webkit-test-runner [ CoreMathMLEnabled=false ] -->
```
Can we rewrite the test to make it work independently of the flag?

Does any of the following options work:
- remove the `csymbol` so `annotation-xml` becomes the first child and is rendered.
- add some CSS rules to explicitly make `annotation-xml` visible (probably only works if MathML Core is enabled).
- use `<math id="math"><mtext><svg height="30"> width="75">...</svg></mtext></math>` instead.

eerii replied:
I used `mtext` as it seemed more reliable. I'll have to wait for EWS to finish to get the updated expectations for Mac and Windows, since I think the trees might be different without the nesting.

fred-wang reviewed (changes requested):
Code change LGTM, but I have some questions about tests.

eerii committed:
Align semantics and maction with MathML Core
https://bugs.webkit.org/show_bug.cgi?id=262696

Reviewed by Frédéric Wang.

The MathML Core spec says that semantics
(https://w3c.github.io/mathml-core/#semantics-and-presentation) and
maction (https://w3c.github.io/mathml-core/#dfn-maction) have the same
layout algorithm as the mrow element. Additionally, add some rules to
the UA stylesheet to properly render these elements
(https://w3c.github.io/mathml-core/#user-agent-stylesheet).

Imports the /mathml/relations/html5-tree/maction-click.html test from
the WPT upstream repository.

Upstream commit: web-platform-tests/wpt@34e2cc1

Tests: imported/w3c/web-platform-tests/mathml/relations/html5-tree/maction-click-ref.html
       imported/w3c/web-platform-tests/mathml/relations/html5-tree/maction-click.html
       mathml/non-core/maction-dynamic.html
       mathml/non-core/maction-toggle-href.html
       mathml/non-core/maction-toggle.html
       mathml/non-core/maction.html
       mathml/non-core/semantics-2.html
       mathml/non-core/semantics-3.html
       mathml/non-core/semantics-4.html
       mathml/non-core/semantics.html
* LayoutTests/accessibility/math-foreign-content.html:
* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/html5-tree/maction-click-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/html5-tree/maction-click-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/html5-tree/maction-click.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/html5-tree/w3c-import.log:
* LayoutTests/mathml/non-core/maction-dynamic-expected.html: Renamed from LayoutTests/mathml/presentation/maction-dynamic-expected.html.
* LayoutTests/mathml/non-core/maction-dynamic.html: Renamed from LayoutTests/mathml/presentation/maction-dynamic.html.
* LayoutTests/mathml/non-core/maction-expected.html: Renamed from LayoutTests/mathml/presentation/maction-expected.html.
* LayoutTests/mathml/non-core/maction-toggle-expected.html: Renamed from LayoutTests/mathml/presentation/maction-toggle-expected.html.
* LayoutTests/mathml/non-core/maction-toggle-href-expected.html: Renamed from LayoutTests/mathml/presentation/maction-toggle-href-expected.html.
* LayoutTests/mathml/non-core/maction-toggle-href.html: Renamed from LayoutTests/mathml/presentation/maction-toggle-href.html.
* LayoutTests/mathml/non-core/maction-toggle.html: Renamed from LayoutTests/mathml/presentation/maction-toggle.html.
* LayoutTests/mathml/non-core/maction.html: Renamed from LayoutTests/mathml/presentation/maction.html.
* LayoutTests/mathml/non-core/semantics-2-expected.html: Renamed from LayoutTests/mathml/presentation/semantics-2-expected.html.
* LayoutTests/mathml/non-core/semantics-2.html: Renamed from LayoutTests/mathml/presentation/semantics-2.html.
* LayoutTests/mathml/non-core/semantics-3-expected.html: Renamed from LayoutTests/mathml/presentation/semantics-3-expected.html.
* LayoutTests/mathml/non-core/semantics-3.html: Renamed from LayoutTests/mathml/presentation/semantics-3.html.
* LayoutTests/mathml/non-core/semantics-4-expected.html: Renamed from LayoutTests/mathml/presentation/semantics-4-expected.html.
* LayoutTests/mathml/non-core/semantics-4.html: Renamed from LayoutTests/mathml/presentation/semantics-4.html.
* LayoutTests/mathml/non-core/semantics-expected.html: Renamed from LayoutTests/mathml/presentation/semantics-expected.html.
* LayoutTests/mathml/non-core/semantics.html: Renamed from LayoutTests/mathml/presentation/semantics.html.
* LayoutTests/platform/glib/accessibility/math-foreign-content-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/html/syntax/parsing/html-integration-point-expected.txt: Added.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/presentation-markup/mrow/legacy-mrow-like-elements-001-expected.txt: Added.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/relations/css-styling/default-properties-on-semantics-and-maction-expected.txt: Added.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/relations/css-styling/ignored-properties-001-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/relations/css-styling/writing-mode/writing-mode-002-expected.txt: Added.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/relations/html5-tree/dynamic-childlist-002-expected.txt:
* LayoutTests/platform/glib/mathml/presentation/foreign-element-in-annotation-xml-expected.txt: Added.
* LayoutTests/platform/mac/accessibility/math-foreign-content-expected.txt:
* LayoutTests/svg/outermost-svg-root.html:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/css/mathmlCoreExtras.css: Added.
(semantics > :not(:first-child)):
(maction > :not(:first-child)):
* Source/WebCore/mathml/MathMLAnnotationElement.cpp:
(WebCore::MathMLAnnotationElement::createElementRenderer):
(WebCore::MathMLAnnotationElement::childShouldCreateRenderer const):
* Source/WebCore/mathml/MathMLSelectElement.cpp:
(WebCore::MathMLSelectElement::childShouldCreateRenderer const):
(WebCore::MathMLSelectElement::updateSelectedChild):
(WebCore::MathMLSelectElement::defaultEventHandler):
(WebCore::MathMLSelectElement::willRespondToMouseClickEventsWithEditability const):
* Source/WebCore/style/InspectorCSSOMWrappers.cpp:
(WebCore::Style::InspectorCSSOMWrappers::collectDocumentWrappers):
* Source/WebCore/style/UserAgentStyle.cpp:
(WebCore::Style::UserAgentStyle::ensureDefaultStyleSheetsForElement):
* Source/WebCore/style/UserAgentStyle.h:

Canonical link: https://commits.webkit.org/300875@main
